salt/modules/file.py

```diff
@@ -2378,12 +2378,12 @@ def blockreplace(path,
 
                         # Check for multi-line '\n' terminated content as split will
                         # introduce an unwanted additional new line.
-                        if content and content[-1] == '\n':
+                        if content and content[-1] == os.linesep:
                             content = content[:-1]
 
                         # push new block content in file
-                        for cline in content.split('\n'):
-                            new_file.append(cline + '\n')
+                        for cline in content.split(os.linesep):
+                            new_file.append(cline + os.linesep)
```

Comment:
All of the changes in the above few lines will also break on Python 3 now that this file is read in binary.